# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
+ Formulas for the Bernoulli distribution: https://en.wikipedia.org/wiki/Bernoulli_distribution

In [125]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from scipy import stats

In [126]:
data = pd.io.stata.read_stata('us_job_market_discrimination.dta')

In [127]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [128]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [129]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

#### Question 1. What test is appropriate for this problem? Does CLT apply?

In [130]:
# Find data size
print(len(data))

4870


#### Answer 1. Because we are looking for proportions and have a sample size greater than 30, the appropriate test to use is a Z-test. The CLT applies because of there are 4870 resumes.

#### Question 2. What are the null and alternate hypotheses?

#### Answer 2. The null hypothesis states that there is no difference in the proportion of call backs depending on race. The alternate hypothesis states that there is a differences in the proportion of call backs depending on race 

#### Question 3. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approaches

In [131]:
# Create proportions
w = data[data.race=='w']
b = data[data.race=='b']

white_callbacks = len(w[w.call==1.0])
total_white = len(w.call)
white_proportion = white_callbacks/total_white

black_callbacks = len(b[b.call==1.0])
total_black = len(b.call)
black_proportion = black_callbacks/total_black

diff_mean = np.abs(white_proportion-black_proportion)
pooled_proportion = (white_callbacks+black_callbacks)/(total_white+total_black)

print(white_proportion)
print(black_proportion)

0.09650924024640657
0.06447638603696099


In [132]:
# Standard error and Margin of Error
z_stat = diff_mean / se 
se = np.sqrt(pooled_proportion * (1-pooled_proportion) * ((1/total_black)+(1/total_white)) )
z_critical_99 = stats.norm.ppf(0.995)
moe = z_critical_99 * se

In [133]:
# Confidence interval
# 95% CI
ci_lower_95, ci_upper_95 = stats.norm.interval(0.95, loc=diff_mean, scale=se)

# 99% CI
ci_lower_99, ci_upper_99 = stats.norm.interval(0.99, loc=diff_mean, scale=se)

print(ci_lower_95, ci_upper_95)
print(ci_lower_99, ci_upper_99)

0.01675122270727635 0.04731448571161482
0.011949386074412063 0.052116322344479105


In [134]:
# Z test to find p-value
z_stat = diff_mean / se 
p_val = stats.norm.sf(np.abs(z_stat))*2

print(z_stat)
print(p_val)

4.108412152434346
3.983886837585077e-05


#### Answer 3. The margin of error is 0.02, the 95% confidence interval is (0.017 to 0.047), and the p-value is 0.00004. Compute margin of error, confidence interval, and p-value. Try using both the bootstrapping and the frequentist statistical approachesse

#### Question 4. Write a story describing the statistical significance in the context or the original problem.

#### Answer 4. The goal of this project is to determine if racial bias affects the likelihood of callbacks for resumes. White-sounding resumes are more likely to get callbacks compared to Black-sounding resumes. A proportional Z-test shows that this difference is statistically significant. Subsequently, we can say that racial bias exists in the hiring process.

#### Question 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

#### Answer 5. While the analysis suggests that there may be a racial discrepency in the callback rates, race may not be the main deciding factor in callback success. There are additional parameters that affect callbacks like gender, years of experience, and education levels. For future studies, a logical regression or correlation test could determine which parameter is most closely associated with callback success.